### ISSUE TITLE
TypeAlias bound forward reference to Generic class results in Undefined error

### ISSUE THREAD

**Issue by chet-manley** [microsoft/pyright#8262]
## Environment data

-   Language Server version: 2024.6.1 (pyright version 1.1.364, commit 0618acc5)
-   OS and version: macOS Sonoma 14.5
-   Python version (& distribution if applicable, e.g. Anaconda): CPython 3.11.9

## Code Snippet

```python
from typing import TYPE_CHECKING, Generic, TypeVar

if TYPE_CHECKING:
    from typing import TypeAlias

    Foo: TypeAlias = 'Bar[int]' # <-- error occurs here

T = TypeVar('T')

class Bar(Generic[T]):
    pass
```

## Repro Steps

1. Paste above code snippet into VSCode.

## Expected behavior

No errors or warnings.

## Actual behavior

`"Bar" is not defined` error, and hovering over `Foo` in the definition shows `(type alias) Foo: Any`.  
_However,_ the alias performs exactly as expected when used.

## Logs

```
May post if absolutely required, but would require tons of data sanitization to be placed in public.
```

**Comment by chet-manley** [comment]
Thank you very much from such a swift fix!
